worker/apicaller/open.go

```diff
@@ -47,10 +47,24 @@ func OpenAPIState(a agent.Agent) (_ api.Connection, err error) {
 	if !ok {
 		return nil, errors.New("API info not available")
 	}
+	if info.Password == "" {
+		logger.Debugf("new config with no password, using password sent over wire before generating new one")
+		// This will only happen for the brand new agent.
+		// In this instance, use old password to verify that
+		// we can connect with old password.
+		info.Password = agentConfig.OldPassword()
+	}
+
```

Comment:
this feels a bit odd, why not just check for `info.Password == ""` in openAPIStateUsingInfo?